File: core/core-domain/src/main/java/com/coffee/gu/menu/MenuRepository.java
package com.coffee.gu.menu;

import org.springframework.data.domain.Pageable;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

public interface MenuRepository {
    List<Menu> findByCategoryId(Long categoryId, Pageable pageable);
    List<Menu> findByCategoryId(Long categoryId, LocalDateTime cursor, Long lastId, int pageSize);
    Optional<Menu> findById(Long menuId);
    List<Menu> findAllByIdIn(List<Long> menuIds);
}

File: core/core-domain/src/main/kotlin/com/coffee/gu/menu/MenuRepository.kt
package com.coffee.gu.menu

import com.coffee.gu.OffsetLimit
import java.time.LocalDateTime
import java.util.Optional

interface MenuRepository {
    fun findByCategoryId(categoryId: Long, offsetLimit: OffsetLimit): List<Menu>
    fun findByCategoryId(categoryId: Long, cursor: LocalDateTime?, lastId: Long?, pageSize: Int): List<Menu>
    fun findById(menuId: Long): Menu
    fun findAllByIdIn(menuIds: List<Long>): List<Menu>
}
